instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/patches/consumer.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      module Patches
+        # The Consumer module contains the instrumentation patch for the Consumer class
+        module Consumer
+          def each_message(min_bytes: 1, max_bytes: 10_485_760, max_wait_time: 1, automatically_mark_as_processed: true)
+            super do |message|
+              attributes = { 'messaging.system' => 'kafka' }
+              attributes['messaging.destination'] = message.topic
+              attributes['messaging.kafka.message_key'] = message.key if message.key
+              attributes['messaging.kafka.partition'] = message.partition
+
+              attributes['offset'] = message.offset
+
+              parent_context = OpenTelemetry.propagation.text.extract(message.headers)
+              tracer.in_span('process', with_parent: parent_context, attributes: attributes, kind: :consumer) do
+                yield message
+              end
+            end
+          end
+
+          def each_batch(min_bytes: 1, max_bytes: 10_485_760, max_wait_time: 1, automatically_mark_as_processed: true)
+            super do |batch|
+              attributes = { 'messaging.system' => 'kafka' }
+              attributes['messaging.destination'] = batch.topic
+              attributes['messaging.kafka.partition'] = batch.partition
+
+              attributes['offset_lag'] = batch.offset_lag
+              attributes['highwater_mark_offset'] = batch.highwater_mark_offset
+              attributes['message_count'] = batch.messages.count
```

Comment:
Do all these (and `offset`) come from Datadog's instrumentation?

Reply:
Yes, any attribute not prefixed with `messaging` was pulled from datadogs instrumentation.

Reply:
Should we drop these?

Reply:
I don't know whether they're useful. We could hide them behind a config option. If we keep them (whether optional or not), we should prefix them with `messaging.kafka.`.

Reply:
or just prefix them with `kafka.` to indicate they are not part of the official schema?

Reply:
I've added the `messaging.kafka` prefix so that things remain consistent.  These attributes were lifted from ddtrace, and while I know it's not great to make assumptions, I'm going to err to side of believing these were added because they provide value and we too should include them.